### PR TITLE
API - Bugfix - Host Application Access

### PR DESCRIPTION
**Pull request by shaangill025**
*Issue:*

- https://github.com/bcgov/entity/issues/23279

*Description of changes:*
Security bug is fixed. The API behaviour [in terms of response codes] remains unchanged for now. Only the authorization check function needed updates.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Review by dimak1 — APPROVED** [review]
👍🏻

**Comment by shaangill025** [comment]
image
CI check is failing with the same issue that Patrick had told to disregard earlier this week.

File: strr-api/tests/unit/resources/test_registration_applications.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def test_get_application_ltsa(session, client, jwt):
             application_number=Application.generate_unique_application_number(),
         )
         application.save()
-        headers = create_header(jwt, [STRR_EXAMINER], "Account-Id")
+        headers = create_header(jwt, [STRR_EXAMINER, STAFF_ROLE], "Account-Id")
```
[the remark this answers is not in the document]
Comment (shaangill025) [reply strr-api/tests/unit/resources/test_registration_applications.py:130]: [is_user_authorized_for_application](https://github.com/bcgov/STRR/blob/ce015d4431ed9da3c6421311b8f0b59328fb3683/strr-api/src/strr_api/services/application_service.py#L112) calls [is_examiner](https://github.com/bcgov/STRR/blob/ce015d4431ed9da3c6421311b8f0b59328fb3683/strr-api/src/strr_api/utils/user_context.py#L97) which only check for `STAFF` role only. I believe this was added as part of rfc2 changes. I will make the change as part of `23364` PR, should it be `Role.STAFF.value and Role.STRR_EXAMINER.value in self._roles` or just `Role.STRR_EXAMINER.value in self._roles`.

The bug was always returning `True` so this unit test were passing earlier incorrectly.

File: strr-api/tests/unit/resources/test_registration_applications.py [hunk 2]
```diff
@@ -155,7 +155,7 @@ def test_get_application_auto_approval(session, client, jwt):
...
```
[the remark this answers is not in the document]
Comment (shaangill025) [reply strr-api/tests/unit/resources/test_registration_applications.py:158]: Check above.

File: strr-api/src/strr_api/services/application_service.py [hunk 3]
```diff
@@ -109,9 +109,9 @@ def get_application(application_id, account_id=None, **kwargs):
     def is_user_authorized_for_application(user_id: int, account_id: int, application_id: int, **kwargs) -> bool:
         """Check the user authorization for an application."""
         usr_context: UserContext = kwargs["user_context"]
-        if usr_context.is_examiner() or usr_context.is_system:
+        if usr_context.is_examiner() or usr_context.is_system():
             return True
-        application = Application.get_application(user_id, account_id, application_id)
+        application = Application.find_by_id_and_user(application_id, user_id)
```
[the remark this answers is not in the document]
Comment (shaangill025) [reply strr-api/src/strr_api/services/application_service.py:114]: Can you replicate this in Dev? Please share the applications IDs. I initially looked up applications by `application_id`, `user_id` and `account_id` and it wasn't returning any applications which is incorrect.

File: strr-api/tests/unit/resources/test_registration_applications.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shaangill025) [reply strr-api/tests/unit/resources/test_registration_applications.py:130]: Okay, I will change to `Role.STRR_EXAMINER.value in self._roles`.

[the remark this answers is not in the document]
Comment (shaangill025) [reply strr-api/tests/unit/resources/test_registration_applications.py:130]: FYI, this was added in this commit by you.
https://github.com/bcgov/STRR/commit/c839b4037aeb7e4d46cfdc9100df289d68a990e2#diff-d9bf6190191e5380bdbcf7e9fabae31daa8312ba692bb7742eb36f6c62b012f4

**Pull request merged**